src/comment.rs

```diff
@@ -362,24 +362,36 @@ fn identify_comment(
     };
 
     let (first_group, rest) = orig.split_at(first_group_ending);
-    let rewritten_first_group =
-        if !config.normalize_comments() && has_bare_lines && style.is_block_comment() {
-            trim_left_preserve_layout(first_group, shape.indent, config)?
-        } else if !config.normalize_comments()
-            && !config.wrap_comments()
-            && !config.format_code_in_doc_comments()
-        {
-            light_rewrite_comment(first_group, shape.indent, config, is_doc_comment)
-        } else {
-            rewrite_comment_inner(
-                first_group,
-                block_style,
-                style,
-                shape,
-                config,
-                is_doc_comment || style.is_doc_comment(),
-            )?
-        };
+
+    let is_doc_comment_with_our_determination = is_doc_comment || style.is_doc_comment();
+
+    let should_trim_left_preserve_layout =
+        { !config.normalize_comments() && has_bare_lines && style.is_block_comment() };
+
+    let should_not_just_lightly_rewrite_comment = {
+        config.normalize_comments()
+            || config.wrap_comments()
+            || (
+                // `format_code_in_doc_comments` should only take effect on doc comments,
+                // so we only consider it when this comment block is a doc comment block.
+                is_doc_comment_with_our_determination && config.format_code_in_doc_comments()
+            )
+    };
+
+    let rewritten_first_group = if should_trim_left_preserve_layout {
+        trim_left_preserve_layout(first_group, shape.indent, config)?
+    } else if !should_not_just_lightly_rewrite_comment {
+        light_rewrite_comment(first_group, shape.indent, config, is_doc_comment)
+    } else {
+        rewrite_comment_inner(
+            first_group,
+            block_style,
+            style,
+            shape,
+            config,
+            is_doc_comment_with_our_determination,
+        )?
+    };
```

Comment:
I appreciate you trying to add some descriptive variable names here, but could you please try and minimize the changes that need to be made in order to fix the issue. Correct me if I'm wrong, but I think the following would have resolved the original issue.

```diff
        } else if !config.normalize_comments()
            && !config.wrap_comments()
-            && !config.format_code_in_doc_comments()
+            && (!is_doc_comment || !config.format_code_in_doc_comments())
```

Reply:
Sure. I will just improve my own logic. However, clippy reports `nonminimal_bool` in this PR now, though it is not a fatal issue.

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/28441561/188775960-3519db69-44e6-43d1-b434-60f74fcda0ec.png">

I may open another PR to improve the readability of this logic. Anyway, thanks for your valuable suggestion!